press-sync.php

```diff
@@ -702,6 +709,97 @@ public function approve_localhost_urls( $approve, $host, $url ) {
+
+		$this->progress->start( 'Syncing ' . $content_type . 's', $this->count_objects_to_sync( $content_type ) );
+
+		while ( $do_run ) {
+
+			$response                = $this->sync_batch( $content_type, $settings, $next_page );
+			$total_objects           = isset( $response['total_objects'] ) ? (int) $response['total_objects'] : 0;
+			$total_objects_processed = isset( $response['total_objects_processed'] ) ? (int) $response['total_objects_processed'] : 0;
+			$next_page               = isset( $response['next_page'] ) ? $response['next_page'] : 0;
+
+			if ( $total_objects === $total_objects_processed ) {
+				$do_run = false;
+			}
+
+			$this->progress->tick();
+		}
+
+		$this->progress->finish();
+
+		return $response;
+	}
+
+	/**
+	 * Synchronize a batch of records.
+	 *
+	 * @param string  $content_type The WP object you want to sync.
+	 * @param array   $settings     The Press Sync settings.
+	 * @param integer $next_page    Pagination counter for bath queries.
+	 */
+	public function sync_batch( $content_type = 'post', $settings = array(), $next_page = 1 ) {
+
+		$remote_domain     = isset( $settings['remote_domain'] ) ? $settings['remote_domain'] : '';
+		$press_sync_key    = isset( $settings['remote_press_sync_key'] ) ? $settings['remote_press_sync_key'] : '';
+		$sync_method       = isset( $settings['sync_method'] ) ? $settings['sync_method'] : cmb2_get_option( 'press-sync-options', 'sync_method' );
+		$objects_to_sync   = $content_type ? $content_type : cmb2_get_option( 'press-sync-options', 'objects_to_sync' );
+		$duplicate_action  = isset( $settings['duplicate_action'] ) ? $settings['duplicate_action'] : cmb2_get_option( 'press-sync-options', 'duplicate_action' );
+		$force_update      = isset( $settings['force_update'] ) ? $settings['force_update'] : cmb2_get_option( 'press-sync-options', 'force_update' );
+
+		// Initialize the connection credentials.
+		$this->init_connection( $remote_domain );
+
+		$prepare_object    = ! in_array( $objects_to_sync, array( 'attachment', 'comment', 'user' ), true ) ? 'post' : $objects_to_sync;
+		$wp_object         = in_array( $objects_to_sync, array( 'attachment', 'comment', 'user' ), true ) ? ucwords( $objects_to_sync ) . 's' : get_post_type_object( $objects_to_sync );
+		$wp_object         = isset( $wp_object->labels->name ) ? $wp_object->labels->name : $wp_object;
+
+		// Build out the url.
+		$url               = cmb2_get_option( 'press-sync-options', 'connected_server' );
+		$press_sync_key    = cmb2_get_option( 'press-sync-options', 'remote_press_sync_key' );
+		$url               = untrailingslashit( $url ) . '/wp-json/press-sync/v1/sync?press_sync_key=' . $press_sync_key;
+
+		// Prepare the remote request args.
+		$sync_args['duplicate_action']  = $duplicate_action;
+		$sync_args['force_update']      = $force_update;
+		$sync_args['objects_to_sync']   = $prepare_object;
+
+		// Prepare the correct sync method.
+		$sync_class        = 'prepare_' . $prepare_object . '_args_to_sync';
```

Comment:
Per WP's coding standards, strings should be interpolated when possible:

```
$sync_class = "prepare_{$prepare_object}_args_to_sync";
```

Reply:
Good update!